### PR TITLE
Update to rustls 0.22

### PR DESCRIPTION
**Pull request by jsha**
DO NOT MERGE until rustls 0.22 is actually released. Right now this uses a Cargo.toml patch to get the latest version before it's released.

### CODE REVIEW

File: Cargo.toml [hunk 1]
```diff
@@ -44,8 +44,9 @@ encoding_rs = { version = "0.8", optional = true }
 cookie_store = { version = "0.20", optional = true, default-features = false, features = ["preserve_order"] }
 log = "0.4"
 webpki = { package = "rustls-webpki", version = "0.101", optional = true }
```
Comment (messense) [inline Cargo.toml:46]: Consider also update `rustls-webpki` to 0.102?

**Comment by Revertron** [comment]
`rustls 0.22` has been released!
Can we update ureq? :)
(I need the MIPS support added by rustls)

**Comment by algesten** [comment]
@jsha once it's failing on httpbin calls, you need to keep rerun only the failed tests many times to "grind" it through.

**Comment by cpu** [comment]
>  I'm really scratching my head on what that method should return.

@amunra If you want your `ServerCertVerifier` to act as it did prior to Rustls 0.22 when there was a default impl of `supported_verify_schemes` you can change the body to defer to the *ring* based provider's supported verify schemes:

```rust
fn supported_verify_schemes(&self) -> Vec<rustls::SignatureScheme> {
    rustls::crypto::ring::default_provider()
        .signature_verification_algorithms
        .supported_schemes()
}
```

**Comment by emilk** [comment]
I would appreciate a patch release with this :pray:

**Comment by aeyoll** [comment]
hello @algesten, can you create a new release with this fix please? Thanks a lot!

**Comment by algesten** [comment]
Alright. Let's do it.

**Pull request merged**